mirrord/layer/src/debugger_ports.rs

```diff
@@ -0,0 +1,149 @@
+use std::{
+    env,
+    net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr},
+    ops::RangeInclusive,
+    str::FromStr,
+};
+
+use tracing::{error, warn};
+
+/// Environment variable used to tell the layer that it should dynamically detect the local port
+/// port used by the given debugger. Value passed through this variable should parse into
+/// [`DebuggerType`]. Used when injecting the layer through IDE, because the debugger port is chosen
+/// dynamically.
+pub const MIRRORD_DETECT_DEBUGGER_PORT_ENV: &str = "MIRRORD_DETECT_DEBUGGER_PORT";
+
+/// Environment variable used to tell the layer that it should ignore certain local ports used by
+/// the debugger. Used when injecting the layer through IDE.
+///
+/// Value passed through this variable can represent a single port like '12233' or a range of ports
+/// like `12233-13000`.
+pub const MIRRORD_IGNORE_DEBUGGER_PORTS_ENV: &str = "MIRRORD_IGNORE_DEBUGGER_PORTS";
+
+/// Type of debugger which is used to run the user's processes.
+#[derive(Debug, Clone, Copy)]
+pub enum DebuggerType {
+    DebugPy,
+}
+
+impl FromStr for DebuggerType {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "debugpy" => Ok(Self::DebugPy),
+            _ => Err(format!("invalid debugger type: {s}")),
+        }
+    }
+}
+
+impl DebuggerType {
+    fn get_port(self, args: &[String]) -> Option<u16> {
+        match self {
+            Self::DebugPy => {
+                let is_python = args.first()?.rsplit('/').next().unwrap().contains("python");
```

Comment:
I would put an example of handled string vs not to give better context for other devs.

Reply:
`unwrap` is evil

Reply:
`unwrap` is evil indeed. `split` and `rsplit` always return at least one item, though, so I hesitated. I can change it

Reply:
That's true but we believe in being holier than pope (so we don't crash :) )